frontend/src/modules/alarms/AlarmsPage.tsx

```diff
@@ -292,41 +294,71 @@ export const AlarmsPage = (): JSX.Element => {
     setPage(value);
   };
 
-  const alarmLimitsRequestStandby = useSelector(getAlarmLimitsRequestStandby);
+  const alarmLimitsRequestDraft = useSelector(getAlarmLimitsRequestDraft);
+  const alarmLimitsCurrent = useSelector(getAlarmLimitsCurrent);
   const dispatch = useDispatch();
   const currentMode = useSelector(getParametersRequestMode);
   const ventilating = useSelector(getParametersIsVentilating);
-  const setAlarmLimits = (data: Partial<AlarmLimitsRequest>) => {
+  const alarmLimitsUnsaved = useSelector(getAlarmLimitsUnsavedChanges);
+  const [alarmLimits] = useState((alarmLimitsCurrent as unknown) as Record<string, Range>);
+  const alarmLimitsDraft = (alarmLimitsRequestDraft as unknown) as Record<string, Range>;
+  const setAlarmLimitsRequestDraft = (data: Partial<AlarmLimitsRequest>) => {
     dispatch(commitStandbyRequest<AlarmLimitsRequest>(MessageType.AlarmLimitsRequest, data));
   };
   const applyChanges = () => {
-    if (alarmLimitsRequestStandby === null) {
+    if (alarmLimitsRequestDraft === null) {
       return;
     }
 
     dispatch(
-      commitRequest<AlarmLimitsRequest>(MessageType.AlarmLimitsRequest, alarmLimitsRequestStandby),
+      commitRequest<AlarmLimitsRequest>(MessageType.AlarmLimitsRequest, alarmLimitsRequestDraft),
     );
   };
   const alarmConfig = alarmConfiguration(currentMode);
   const [open, setOpen] = useState(false);
+  const [discardOpen, setDiscardOpen] = useState(false);
+  const [isDisabled, setIsDisabled] = useState(!alarmLimitsUnsaved);
 
   const handleClose = () => {
     setOpen(false);
   };
 
+  const handleDiscardClose = () => {
+    setDiscardOpen(false);
+  };
+
   const handleConfirm = () => {
     setOpen(false);
     applyChanges();
+    setIsDisabled(true);
+  };
+
+  const handleDiscardConfirm = () => {
+    setDiscardOpen(false);
+    setAlarmLimitsRequestDraft(alarmLimits);
+    setIsDisabled(true);
```

Comment:
If the `disabled` property of those buttons is actually determined by a selector (perhaps `getAlarmLimitsUnsavedChanges`, which I haven't reviewed yet) which tells whether there are any differences between `getAlarmLimitsRequest` and `getAlarmLimitsRequestDraft`, then we can get rid of the `isDisabled, setIsDisabled` useState, right? That way we can guarantee that even if changes are made elsewhere to `store.controller.alarmLimits.request`, such as from the breadcrumbs modal via the Events Log modal, the Cancel/Submit buttons will always be in the correct state, and we don't need to even think about the possibility of contention/overwrites between this `setIsDisabled` call here and the `useEffect` below synchronizing `setIsDisabled` with `alarmLimitsUnsaved`.

Reply:
yep I agree, using the selector now and removed the useEffect